BLSQ/iaso
# Polio 1358 vaccine module format thousand

We needed number formatting for vaccine supplychain

Related JIRA tickets : POLIO-1358

## Self proofreading checklist


## Changes

- Import `NumberInput` and `FormattedNumber` from `bluesquare-components`
- Refactor `InputComponent` to use `NumberInput`. Add props and get thousand and decimal separators from the locale by default
- Add `NumberCell` which uses `FormattedNumber` with the same defaults as `InputComponent` but can be used in tables and text.
- Update all components in polio that used `TextInput` to show numbers so they use `NumberInput` instead
- Update Vaccine puppy chain and vaccine stock management
- Harmonize layout in polio campaigns modal: empty fields don't shrink the label anymore


## How to test
- run `npm ci`
- replace dep to bluesquare-components in package.json :
`"bluesquare-components": "github:BLSQ/bluesquare-components#POLIO-1358_thousand_separator",`
- run `npm update bluesquare-components && npm run dev` (you will run webpack outside of docker)
- Go through iaso and check that numbers are formatted where needed. 


What you should see: when setting language to english, thousands should be separated by commas, and decimals by a dot. When setting language to French it should be the opposite.

So when checking high integers, check in english, when checking floats (like latitude/longitude), check in french.

## Print screen / video


https://github.com/BLSQ/iaso/assets/38907762/561121e6-9b7b-47de-912f-d75ed5c983c7



## Notes
⚠️Depends on https://github.com/BLSQ/bluesquare-components/pull/137

TESTS WILL FAIL UNTIL https://github.com/BLSQ/bluesquare-components/pull/137 is merged and the dep updated
